packages/webapp/src/containers/Task/TaskLocations/index.jsx

```diff
@@ -166,7 +167,44 @@ function TaskAnimalLocations({ history, location }) {
   );
 }
 
-function TaskAllLocations({ history, location, optionalLocation }) {
+function TaskCustomLocations({ history, location }) {
```

Comment:
Looks like the logic in this one is pretty much the same as in `TaskAllLocations` except for the bit where it checks for animals. Do you think it'd be a good idea to have this be an option in `TaskAllLocations` instead? Something like `includeAnimalSelection` so we could reuse this in the future for system tasks related to animals that have this same flow

```js
function TaskAllLocations({ history, location, includeAnimalSelection = false }) {
  const dispatch = useDispatch();
  const locations = useSelector(locationsSelector);
  const persistedFormData = useSelector(hookFormPersistSelector);
  const taskTypesBypassCrops = useSelector(taskTypeIdNoCropsSelector);
  const readOnlyPinCoordinates = useReadOnlyPinCoordinates();
  const activeAndCurrentManagementPlansByLocationIds =
    useActiveAndCurrentManagementPlanTilesByLocationIds(locations);
  const wildManagementPlanTiles = useCurrentWildManagementPlanTiles();
  const animalsExistOnFarm = useAnimalInventory().inventory.length > 0;
  const showAnimalSelectionStep = includeAnimalSelection && animalsExistOnFarm;

  const onContinue = (formData) => {
    const hasLocationManagementPlans = formData.locations.some(
      ({ location_id }) => activeAndCurrentManagementPlansByLocationIds[location_id]?.length,
    );
    const hasWildManagementPlans = formData.show_wild_crop && wildManagementPlanTiles.length;
    const hasManagementPlans = hasLocationManagementPlans || hasWildManagementPlans;
    if (
      (taskTypesBypassCrops.includes(persistedFormData.task_type_id) &&
        !readOnlyPinCoordinates?.length) ||
      !hasManagementPlans
    ) {
      dispatch(setManagementPlansData([]));
      if (showAnimalSelectionStep) {
        return history.push('/add_task/task_animal_selection', location?.state);
      }
      return history.push('/add_task/task_details', location?.state);
    }
    history.push('/add_task/task_crops', location?.state);
  };
  return (
    <TaskLocations
      locations={locations}
      history={history}
      onContinue={onContinue}
      readOnlyPinCoordinates={readOnlyPinCoordinates}
      location={location}
    />
  );
}
```

Reply:
I originally tried that, but a few things made me go this route. 
- AnimalsInventory wont necessarily be in the same order across tasks (Movement is before location, Custom is after crops)
- I was prop drilling two different animals booleans through Task All locations. 
- I was thinking I could avoid running a few unnecessary selectors

Reply:
Makes sense!! We'll also be able to figure it out better in the long run, once we add more system tasks we'll have a better idea about whether there's a common logic that we'd be able to reuse or not so we could always combine them later if needed.